fix: resolve 50% session tracking discrepancy on root route

seanoliver opened the pull request:
## Summary
- Fixes race condition causing ~50% discrepancy in session tracking between client and server on root route
- Implements event queuing to capture telemetry events before PostHog initializes
- Adds session ID correlation to enable unified tracking analysis between client and server events

Resolves GROWTH-461

## Problem
We observed a discrepancy in unique session counts on the root route only:
- Client-side tracking was missing ~50% of sessions compared to server-side
- This was only happening on the root route, which uses a lot of dynamic imports compared to other pages
- Hypothesis is that users were bouncing before PostHog client library initialized
- This hypothesis can't be fully validated because we don't currently share a unique session id between client and server

<img width="767" height="446" alt="CleanShot 2025-08-11 at 17 59 33" src="https://github.com/user-attachments/assets/e330c114-6dbe-4b72-b479-bd507147fdf7" />

<img width="1730" height="944" alt="CleanShot 2025-08-11 at 18 25 53@2x" src="https://github.com/user-attachments/assets/f2d44fd9-d13f-45be-aeed-bf3724bc9ca3" />

## Solution
1. **Event Queuing**: Queue PV events while PostHog initializes, preventing data loss from early user interactions
2. **Session Correlation**: Extract PostHog session ID from cookies and sync between client/server for unified tracking
3. **Reliable Page Leave**: Use `sendBeacon` API to ensure page-leave events are captured even during navigation
4. **Production Safeguards**: Add `try`/`catch` blocks and memory caps to prevent API errors from breaking our tracking

## Test Plan
- [x] Verify events are queued and sent after PostHog initializes
- [x] Confirm session IDs match between client and server telemetry
- [x] Test page-leave events are captured on navigation
- [x] Vercel preview URLs added to UserCentrics for testing

pamelachia reviewed (approved):
i didn't get any lag when doing identify and event sending so not sure how to test that the queueing thing works but looks good to me otherwise. events do get sent to posthog.

seanoliver committed:
fix: resolve 50% session tracking discrepancy on root route

- Fixes race condition causing ~50% discrepancy in session tracking between client and server on root route
- Implements event queuing to capture telemetry events before PostHog initializes
- Adds session ID correlation to enable unified tracking analysis between client and server events
- Uses sendBeacon API for both pageview and pageleave events to ensure reliable capture even during navigation
- Adds production safeguards with try/catch blocks and memory caps to prevent API errors from breaking tracking

joshenlim reviewed (approved):
Code wise generally LGTM - although as per Pam's comment above, not sure how to test this one locally / on staging, approving to unblock although do address Pam's comments before merging! 🙏 🙂

seanoliver committed:
refactor(posthog-client): reduce max pending events and improve event handling

- Decreased the maximum number of queued events from 200 to 20
- Added the use of the sendBeacon API for pending events to survive page unload
- Added comments to clarify the handling of pending properties and events during posthog initialization
